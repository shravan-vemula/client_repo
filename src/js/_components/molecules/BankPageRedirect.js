import React from 'react'
import { Button } from '@material-ui/core';
import {useLocation} from 'react-router-dom';
import AnimationModalSuccess from 'js/_components/atoms/AnimationModalSuccess'



function BankPageRedirect() {
    const location = useLocation();
    const {url} = location.state;
    const [onUnloadWindow,setOnUnloadWindow] = React.useState(false);

    const handleClick = () =>{
       const myWindow = window.open(url, '_blank', 'location=yes,height=570,width=520,scrollbars=yes,status=yes',false)
  
       myWindow.addEventListener('unload',function(e){
           e.preventDefault();
            console.log(e)
            console.log("unloaded")
            setOnUnloadWindow(true);
              
        })

    }

    if(onUnloadWindow){
        return(
            <>
            <BankPageRedirect />
            <AnimationModalSuccess placeHolder={"Linking Bank"}/>    
            </>
        );
    }

    return (
        <>
    <div id="bank-selection-layout-1" className="bankSelectionLayout1">
        <div id="bank-selection-header" className="bankSelectionHeader">
            <div id="bank-selection-header-text-1"className="bankSelectionHeader1" >
                <p>Connect your account</p>
                </div>
            <div id="bank-selection-header-text-1"className="bankSelectionHeader2" >
                <p><b>Start by connecting your Chase Bank accounts with Intuit, the makers of Mint.</b><br /> <br />
                Go to the bank site to sign in and connect your accounts. Then weâ€™ll return you here and bring<br /> 
                in your transactions.</p>
            </div>  
            <div id="optional-selection" className="optionalSelection">
            <Button data-testId="continueButton" variant="outlined" color="primary" 
            onClick={handleClick}>
                Continue
            </Button>
            </div>  
        </div>
    </div>
    </>
    )
}

export default BankPageRedirect
