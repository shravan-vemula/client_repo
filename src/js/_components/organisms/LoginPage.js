import React from 'react';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import Link from '@material-ui/core/Link';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import {useLocation} from 'react-router-dom';

import AnimationModalSuccess from 'js/_components/atoms/AnimationModalSuccess'


const useStyles = makeStyles((theme) => ({
    paper: {
      marginTop: theme.spacing(8),
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
    },
    avatar: {
      margin: theme.spacing(1),
      backgroundColor: theme.palette.secondary.main,
    },
    form: {
      width: '60%', // Fix IE 11 issue.

      marginTop: theme.spacing(1),
    },
    submit: {
      margin: theme.spacing(3, 0, 2),
    },
  }));

 function LoginPage(props) {
  const [onChangeValue,setOnChangeValue] = React.useState(false)
    const classes = useStyles();
    const location = useLocation();
    const {logo,placeHolder} = location.state

    const handleSubmit = (event) =>{
      event.preventDefault();
      const email = document.getElementById("email").value;
      const password = document.getElementById("password").value;
      
      if(email==='kaushik@gmail.com' && password === 'kaushik'){
        console.log("email is " + email + " password is "+password)
        setOnChangeValue(true);

    }
      else
        alert("Invalid login details");
      
    }

    if(onChangeValue === true){
     
      console.log("true");
     return ( 
       <>
      <LoginPage />
      <AnimationModalSuccess placeHolder={placeHolder}/>
      </>   
      )
    }
    

    return (
      <div id="bank-selection-layout-1" className="bankSelectionLayout1"  style={{fontFamily:"Inter,sans-serif"}}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <div className={classes.paper}  style={{fontFamily:"Inter,sans-serif"}}>
        <Avatar alt="Remy Sharp" src={logo} className={classes.large} variant="square"/>
          <Typography component="h1" variant="h5"  style={{fontFamily:"Inter,sans-serif"}}>
            Sign in
          </Typography>
          <form className={classes.form} data-testId="LoginSubmit" noValidate onSubmit={handleSubmit}>
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              inputProps={{ "data-testid": "username" }}
              id="email"
              label= {placeHolder}
              name="email"
              autoComplete="email"
              autoFocus
            />
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              inputProps={{ "data-testid": "password" }}
              name="password"
              label="Password"
              type="password"
              id="password"
              autoComplete="current-password"
            />
            <FormControlLabel
              control={<Checkbox value="remember" color="primary" />}
              label="Remember me"
            />
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              data-testId="SigninButton"
              className={classes.submit}
            >
              Sign In
            </Button>
            <Grid container>
              <Grid item xs>
                <Link href="#" variant="body2">
                  Forgot password?
                </Link>
              </Grid>
              <Grid item>
                <Link href="#" variant="body2">
                  {"Don't have an account? Sign Up"}
                </Link>
              </Grid>
            </Grid>
          </form>
          

          
        </div>
      </Container>
      </div>
    );
  }
export default LoginPage
