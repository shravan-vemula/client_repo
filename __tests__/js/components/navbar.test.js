import React from 'react';
import NavBar from 'js/_components/organisms/Navbar';
import { render, screen, fireEvent, getByRole, getByTestId } from '@testing-library/react';
import {BrowserRouter as Router, MemoryRouter} from 'react-router-dom';
import BankPageRedirect from 'js/_components/molecules/BankPageRedirect';
import { createMemoryHistory } from "history"
import LoginPage from 'js/_components/organisms/LoginPage';




describe('Navbar renders correctly', () =>{
    test('renders form layout', () =>{
       const {getByTestId} = render(<Router><NavBar /></Router>);
        expect(screen.getByRole('img')).toBe;
        fireEvent.click(getByTestId('LogoImg'))
        screen.debug();
    })
    it("navbar snap shot", () => {
        const wrapper = shallow(<MemoryRouter><BankPageRedirect /> </MemoryRouter>);
        expect(wrapper).toMatchSnapshot();
      });

    
});
