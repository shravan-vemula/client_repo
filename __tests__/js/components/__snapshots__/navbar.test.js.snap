// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly navbar snap shot 1`] = `ShallowWrapper {}`;
